### PR TITLE
SAK-29513 close streams in finally blocks

### PR DESCRIPTION
**Pull request by ottenhoff**
@clhedrick there is one lessons change in here

@ern look okay to you?

### CODE REVIEW

File: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java [hunk 1]
```diff
@@ -4348,6 +4348,14 @@ public void doUpload_Mtrl_Frm_File(RunData data) {
 			catch (IOException ioe) {
 				M_log.warn("IOException creating temp import file",ioe);
 			}
+			finally {
+				try {
+					fileInputStream.close();
```
Comment (buckett) [inline site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java]: Are we sure that fileInputStream will never be null?

**Comment by buckett** [comment]
I'm not suggesting you re-work this, but in future we could use try-with-resources: 

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Although we can't then backport to 10.x without re-working.

**Comment by ottenhoff** [comment]
Okay, I added the null check.

try with resources is a much cleaner pattern.... but i'm hoping to backport these

**Comment by steveswinsburg** [comment]
This is fine, but if not using the try-with-resources (I had to double check we didn't require java 7 for Sakai 10, that was a silly move on our behalf!) I'd use IOUtils.closeQuietly(is) which avoids the null checks.

**Comment by buckett** [comment]
http://archive.apache.org/dist/commons/io/RELEASE-NOTES.txt says 2.0 -> 2.4 is binary compatible (rare source code issue only, https://issues.apache.org/jira/browse/IO-318 ).

Good to merge.

**Pull request merged**